core/src/test/java/com/linecorp/armeria/client/IgnoreHostsTrustManagerTest.java

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.security.cert.X509Certificate;
+import java.util.HashSet;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLEngineResult;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerPort;
+
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+
+class IgnoreHostsTrustManagerTest {
+
+    private static int port;
+    private static Server server;
+    private static Socket defaultSocket;
+    private static SSLEngine defaultSslEngine;
+    private static X509Certificate[] defaultCerts;
+
+    @BeforeAll
+    static void init() {
+        defaultCerts = new X509Certificate[0];
+        defaultSocket = new Socket();
+        defaultSslEngine = new MockSSLEngine("localhost", 0);
+        try {
+            final SelfSignedCertificate ssc = new SelfSignedCertificate("localhost");
+            server = Server.builder()
+                    .service("/", (ctx, req) -> HttpResponse.of())
+                    .tls(ssc.certificate(), ssc.privateKey())
+                    .build();
+            server.start().get();
+            port = server.activePorts().values().stream()
+                    .filter(ServerPort::hasHttps).findAny().get().localAddress()
+                    .getPort();
```

Comment:
Could use `ServerExtension`

Reply:
I noticed it could be used in `HttpClientSniTest` as well so I rewrote this using `ServerExtension`.
https://github.com/line/armeria/blob/198e2d99d3c8b180b59a41cf2746c5320810c22d/core/src/test/java/com/linecorp/armeria/client/HttpClientSniTest.java#L54-L58